app/src/organisms/AppSettings/__tests__/ConnectRobotSlideout.test.tsx

```diff
@@ -0,0 +1,237 @@
+import * as React from 'react'
+import { renderWithProviders } from '@opentrons/components'
+import { fireEvent } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { i18n } from '../../../i18n'
+import { getScanning, getViewableRobots } from '../../../redux/discovery'
+import { getConfig } from '../../../redux/config'
+import { ConnectRobotSlideoutComponent } from '../ConnectRobotSlideout'
+
+jest.mock('../../../redux/discovery')
+jest.mock('../../../redux/config')
+
+const render = (
+  props: React.ComponentProps<typeof ConnectRobotSlideoutComponent>
+) => {
+  return renderWithProviders(<ConnectRobotSlideoutComponent {...props} />, {
+    i18nInstance: i18n,
+  })[0]
+}
+
+const mockGetScanning = getScanning as jest.MockedFunction<typeof getScanning>
+const mockGetConfig = getConfig as jest.MockedFunction<typeof getConfig>
+const mockGetViewableRobots = getViewableRobots as jest.MockedFunction<
+  typeof getViewableRobots
+>
+
+describe('ConnectRobotSlideout', () => {
+  let props: React.ComponentProps<typeof ConnectRobotSlideoutComponent>
+
+  beforeEach(() => {
+    mockGetScanning.mockReturnValue(true)
+
+    mockGetConfig.mockReturnValue({
+      discovery: {
+        candidates: ['1.1.1.1', 'localhost', '192.168.1.1'],
+      },
+    } as any)
+    mockGetViewableRobots.mockReturnValue([
+      {
+        name: 'other-robot-name',
+        host: '1.1.1.1',
+        port: 31950,
+      },
+      {
+        name: 'test-robot-name',
+        host: 'localhost',
+        port: 31950,
+      },
+      {
+        name: 'opentrons',
+        ip: '192.168.1.1',
+        port: 31950,
+        local: false,
+        ok: false,
+        serverOk: true,
+      },
+    ] as any[])
+
+    props = {
+      candidates: [],
+      checkIpAndHostname: jest.fn(),
+      isExpanded: true,
+      onCloseClick: jest.fn(),
+    }
```

Comment:
```suggestion
    } as ConnectRobotSlideoutProps
```
If you add this to each instance of props in this file, you should be able to get rid of the current TS errors.